_posts/2018-10-21-nabla-k8s.md

```diff
@@ -0,0 +1,273 @@
+```
+runnc: v0.2
+containerd: v1.2.0-rc1
+kubernetes: v1.11.0
+```
+
+##  Build and install Nabla runtime `runnc`
+
+
+The first step to running nabla on kubernetes is to run nabla! The first order of business is to make sure we have a running new (v0.2) version of `runnc`. For a step-by-step guide to install `runnc`, take a look at our previous article: [Running a Nabla Container]({% post_url 2018-06-28-nabla-setup %}).
+
+**WARNING: Be sure to use latest commit!**
+
+## Setup containerd with `runnc`
+
+We are using [containerd](https://github.com/containerd/containerd)  as a runtime as it supports the selection of an alternative runtime via a pod annotation in kubernetes. This feature is a cri plugin called `untrusted_workload_runtime`.
+
+We enable this feature by modifying a containerd configuration to include the cri plugin:
+
+```
+lumjjb@lumjjb-ThinkPad-P50:~$ cat /etc/containerd/config.toml
+
+subreaper = true
+oom_score = -999
+
+[debug]
+        level = "debug"
+
+[metrics]
+        address = "127.0.0.1:1338"
+
+[plugins.linux]
+        runtime = "runc"
+        shim_debug = true
+
+### ADD THE FOLLOWING CONFIGURATION ###
+[plugins]
+  [plugins.cri.containerd]
+  [plugins.cri.containerd.untrusted_workload_runtime]
+  runtime_type = "io.containerd.runtime.v1.linux"
+  runtime_engine = "/usr/local/bin/runnc"
+```
+
+We then run the containerd service in preparation for setting up kubernetes (Note: This may require running with `root` or `sudo`):
+
+```
+root@lumjjb-ThinkPad-P50:/home/lumjjb# /usr/local/bin/containerd --config /etc/containerd/config.toml
+
+INFO[2018-10-22T21:40:27.299644455-04:00] starting containerd                           revision=de4bb2ddfbb6b2b6a112b3478a935ca74dd7b796 version=v1.2.0-rc.0-44-gde4bb2d
+DEBU[2018-10-22T21:40:27.299684880-04:00] changing OOM score to -999
+INFO[2018-10-22T21:40:27.299892674-04:00] loading plugin "io.containerd.content.v1.content"...  type=io.containerd.content.v1
+INFO[2018-10-22T21:40:27.299909914-04:00] loading plugin "io.containerd.snapshotter.v1.btrfs"...  type=io.containerd.snapshotter.v1
+<<< TRUNCATED >>>
+```
+
+Note that this process runs as a daemon, so we will need to keep this running for
+the rest of the setup. Optionally, you may set it up as a [systemd service](https://manpages.debian.org/stretch/systemd/systemctl.1.en.html).
+
+## Setting up a CNI config
+
+Note: if you already have CNI configured, you may skip this step.
+
+The [Container Network Interface (CNI)](https://github.com/containernetworking/cni) is the interface that kubernetes uses for networking. A configuration is required for our setup using `hack/local-up-cluster.sh`. For simplicity, we use a simple bridge CNI plugin. 
+
+This can be done by adding the following configuration in the following two files (`/etc/cni/net.d/10-mynet.conflist`, `/etc/cni/net.d/99-loopback.conf`). This configuration will create a bridge with name `cni0` and use it to connect the pods created in the kubernetes cluster.
```

Comment:
This is done before installing kubernetes right? if that's the case, we have to create the dir `/etc/cni/net.d`?

Reply:
Yup - I'll add in that step!

Reply:
Hmm, the reason why i didnt put in a direct overwrite of the file, is because they may already have a CNI configuration in place, and I don't want to mess it up.